modules/programs/ssh.nix

```diff
@@ -0,0 +1,155 @@
+          original X11 display.
+        '';
+      };
+
+      identitiesOnly = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Specifies that ssh should only use the authentication identity
+          explicitly configured in the .ssh/config files or passed on the
+          ssh command-line, even if ssh-agent offers more identities.
+        '';
+      };
+
+      identityFile = mkOption {
+        type = types.nullOr types.string;
+        default = null;
+        description = ''
+          Specifies a file from which user's identity is read.
+        '';
+      };
+
+      user = mkOption {
+        type = types.nullOr types.string;
+        default = null;
+        description = "Specifies the user to log in as.";
+      };
+
+      hostname = mkOption {
+        type = types.nullOr types.string;
+        default = null;
+        description = "Specifies the real host name to log into.";
+      };
+
+      serverAliveInterval = mkOption {
+        type = types.int;
+        default = 0;
+        description =
+          "Set timeout in seconds after which response will be requested.";
+      };
+
+      checkHostIP = mkOption {
+        type = types.bool;
+        default = true;
+        description = "Check that host IP address in the known_hosts file.";
+      };
+    };
+  });
+
+  hostStr = host: cf: concatStringsSep "\n" (
+    ["host ${host}"]
+    ++ optionals (cf.port != null)         ["  port ${toString cf.port}"]
+    ++ optionals cf.forwardX11             ["  forwardX11 yes"]
+    ++ optionals cf.forwardX11Trusted      ["  forwardX11Trusted yes"]
+    ++ optionals cf.identitiesOnly         ["  identitiesOnly yes"]
+    ++ optionals (cf.user != null)         ["  user ${cf.user}"]
+    ++ optionals (cf.identityFile != null) ["  identityFile ${cf.identityFile}"]
+    ++ optionals (cf.hostname != null)     ["  hostname ${cf.hostname}"]
+    ++ optionals (cf.serverAliveInterval != 0)
+         ["  serverAliveInterval ${toString cf.serverAliveInterval}"]
+    ++ optionals (!cf.checkHostIP)         ["  checkHostIP no"]
+  );
+
+in
+
+{
+  options.programs.ssh = {
+    enable = mkEnableOption "SSH Client Configuration";
+
+    forwardAgent = mkOption {
+      default = false;
+      type = types.bool;
+      description = ''
+        Whether connection to authentication agent (if any) will be forwarded
+        to remote machine.
+      '';
+    };
+
+    controlMaster = mkOption {
+      default = "no";
+      type = types.enum ["yes" "no" "ask" "auto" "autoask"];
+      description = ''
+        Configure sharing of multiple sessions over a single network connection.
+      '';
+    };
+
+    controlPath = mkOption {
+      type = types.string;
+      default = "~/.ssh/master-%r@%h:%p";
+      description = ''
+        Specify path to the control socket used for connection sharing.
+      '';
+    };
+
+    hosts = mkOption {
+      type = types.loaOf hostModule;
```

Comment:
I think this shoud be `types.attrsOf` rather than `types.loaOf` since the attribute name is important, i.e., I think it wouldn't work very well if you actually put a list of attributes here.

(I realize I've done the same mistake in the `gnome-terminal` module, will make a commit to fix that)